controllers/user.controller.js

```diff
@@ -225,16 +227,25 @@ export const createAdmin = async (req, res) => {
 };
 export const deleteUser = async (req, res) => {
     const user = res.locals.user;
-    const id = req.body.id;
+    const id = req.params.id;
     try {
         if (user.admin) {
-            await POI.deleteMany({user: id})
-            await User.findByIdAndRemove(id);
-            return res.status(200).json({
-                message: 'User Deleted successfully',
+            const founduser = await User.findOne({
+                _id: new mongoose.Types.ObjectId(id),
             });
```

Comment:
This can be replaced with `findById` to simplify it removing the need for implicitly casting to an ObjectID, `findById` handles that